### DISCUSSION TITLE
Type issues with F# 8.0

### DISCUSSION THREAD

**Discussion by MaxWilson** [dotnet/fsharp#16287] [Q&A]
According to the F# 8 announcement at https://devblogs.microsoft.com/dotnet/announcing-fsharp-8/,

_The same syntax can be used to define accessors via [SRTP syntax](https://learn.microsoft.com/dotnet/fsharp/language-reference/generics/statically-resolved-type-parameters), allowing the same binding to be used by all items having the same member, without the need to share a common interface._

    let inline myPropertyGetter (x: 'a) = x |> _.WhatANiceProperty
    
However, when I try this, I get a type error: **Lookup on object of indeterminate type based on information prior to this program point.**

Similarly, when I try the static let example:

    [<Struct>]
    type MyUnion = 
        | A of aval:'A
        | B of bval:'B
        | C
    
        static let sizeOfTCached = 
            printfn "Creating cached val for %s * %s" (typeof.Name) (typeof.Name)
            sizeof<MyUnion>

I get **error FS0039: The type parameter 'A is not defined.**

Is the blog post wrong or am I doing something wrong? Other F# 8 features like try/with within seq expressions work fine.

**Comment by vzarytovskii** [comment]
First one is missing constraint or specific type. It should include it for type parameter. There's still no auto-inferring for constraints, and likely will never be.

Second one is missing type parameter on type itself.

Examples are wrong.

**Comment by MaxWilson** [comment]
Thanks!
